### PR TITLE
Thin opensds

### PR DESCRIPTION
**Pull request by leonwanghui**
**What this PR does / why we need it**:
This PR is depended on #696 , @Click2Cloud-Gamma please take a review at this patch, thanks!

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/controllers/volume.go [hunk 1]
```diff
@@ -615,10 +779,36 @@ func (v *VolumeSnapshotPortal) CreateVolumeSnapshot() {
...
+	if err != nil {
+		log.Error("create volume snapshot failed in apiserver service:", err)
+		return
+	}
+	if errorMsg := response.GetError(); errorMsg != nil {
+		log.Errorf("failed to create volume snapshot in apiserver, code: %v, message: %v",
+			errorMsg.GetCode(), errorMsg.GetDescription())
 		return
 	}
+	// TODO Update volume snapshot status for Thin OpenSDS in DB
```
Comment (Click2Cloud-Gamma) [inline pkg/api/controllers/volume.go]: Same here

File: pkg/api/controllers/volume.go [hunk 2]
```diff
@@ -747,10 +937,26 @@ func (v *VolumeSnapshotPortal) DeleteVolumeSnapshot() {
...
+	if err != nil {
+		log.Error("delete volume snapshot failed in apiserver service:", err)
 		return
 	}
+	if errorMsg := response.GetError(); errorMsg != nil {
+		log.Errorf("failed to delete volume snapshot in apiserver, code: %v, message: %v",
+			errorMsg.GetCode(), errorMsg.GetDescription())
+		return
+	}
+	// TODO Update volume snapshot status for Thin OpenSDS in DB
```
Comment (Click2Cloud-Gamma) [inline pkg/api/controllers/volume.go]: Same here

**Comment by xing-yang** [comment]
@leonwanghui Discussed with Steven about this.  Let's get this merged into a different repo and release it as an experimental feature for Capri.

**Comment by xing-yang** [comment]
@leonwanghui Please resolve conflicts.  Let's try to get this merged next week.  Thanks.

**Comment by skdwriting** [comment]
@leonwanghui as we discussed let us not merge this to development branch at the moment. Because it will create conflict for other development. Hence, I request to cancel this PR and raise another PR to another branch like 'ThinOpenSDS_beta' OR please wait till we confirm about the strategy. Thanks.

**Comment by leonwanghui** [comment]
Closing this PR, and will raise another PR

**Comment by xing-yang** [comment]
@leonwanghui Can you please submit another PR for this?  We want to get this in Capri as an experimental feature.  Thanks!

**Pull request closed without merging**